shared/origami/CMakeLists.txt

```
@@ -0,0 +1,156 @@
+# Copyright Advanced Micro Devices, Inc., or its affiliates.
+# SPDX-License-Identifier:  MIT
+
+cmake_minimum_required(VERSION 3.24.4)
+
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+    project(Origami VERSION 1.0.0 LANGUAGES CXX)
+
+    set(ORIGAMI_STANDALONE ON)
+else()
+    set(ORIGAMI_STANDALONE OFF)
+endif()
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+include(GenerateExportHeader)
+include(CMakeDependentOption)
+
+include(dependencies)
+
+rocm_setup_version(VERSION ${PROJECT_VERSION})
+
+option(ORIGAMI_BUILD_SHARED_LIBS "Build shared libraries." ON)
+option(ORIGAMI_ENABLE_PYTHON "Enable Python bindings." OFF)
+option(ORIGAMI_BUILD_TESTING "Enable Python binding tests." OFF)
+
+find_package(hip REQUIRED)
+
+if(ORIGAMI_BUILD_SHARED_LIBS)
+    add_library(origami SHARED) 
+else()
+    add_library(origami STATIC)
+endif()
+
+rocm_set_soversion(origami ${PROJECT_VERSION_MAJOR})
+
+add_library(origami::origami ALIAS origami)
+
+set_target_properties(origami PROPERTIES
+    POSITION_INDEPENDENT_CODE ON
+)
+
+target_compile_features(origami PUBLIC cxx_std_17)
+
+target_include_directories(origami
+    PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+)
```

```suggestion
add_library(origami-headers INTERFACE)
add_library(roc::origami-headers ALIAS origami-headers)
target_include_directories(origami
    PUBLIC
        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
)

target_link_libraries(origami PUBLIC roc::origami-headers)
```

Also need a `target_sources` call for the header files.